lib/parse.js

```diff
@@ -520,6 +522,11 @@ function tokenizer($TEXT, filename, html5_comments, shebang) {
         if (RE_OCT_NUMBER.test(num) && next_token.has_directive("use strict")) {
             parse_error("Legacy octal literals are not allowed in strict mode");
         }
+        if (num.endsWith("n")) {
+            if (!has_dot && RE_BIG_INT.test(num))
+                return token("bigInt", parse_js_number(num.replace("n", "")));
```

Comment:
`parse_js_number` uses `parseFloat`, which will round the bigint literal down to the nearest `number`. This causes a loss of precision.

This should be:
```javascript
return token("bigInt", BigInt(num.replace("n", "")));
```

Reply:
Ja.. It works for node >= 10.4.0